docs/reference/query-dsl.asciidoc

```diff
@@ -4,7 +4,7 @@
 [partintro]
 --
 
-Elasticsearch provides a full Query DSL based on JSON to define queries.
+Elasticsearch provides a full Query DSL (Domain Specific Language) based on JSON to define queries.
```

Comment:
I agree that acronyms like DSL are not always common knowledge, and I think adding this here as it is the introduction makes sense. That however raises the question of where to draw the line, should we also explain what JSON and AST in the following line stands for? I'd say maybe no to the former, yes to the later. I'm happy to pull this in here but would also like to hear other thoughts, maybe @clintongormley has an opinion about this as well.

Reply:
Good point. I think it's always hard to decide where to draw the line with these things.

Personally, I would say we wouldn't need an explanation of JSON because of its ubiquity, but AST could also use an `(Abstract Syntax Tree)` note. My reasoning is that I see the Elastic Stack documentation as the entry point for a lot of web developers. I think most web developers know what JSON is, but many of them would never run into the DSL or AST acronyms. They could always look them up on Google, but that goes for everything, and it makes the barrier to entry seem much less intimidating if we see the acronym meanings.

Reply:
I agree - defining terms on their first use is good, but I'd agree that JSON probably doesn't need defining.

Reply:
Thanks, @andrewbanchich would you mind also adding the AST explanation in this case, then I'm going to merge this in.

Reply:
@cbuescher Done. Thanks!